client/dashboard/profile-wizard/steps/industry.js

```diff
@@ -67,16 +69,20 @@ class Industry extends Component {
 	onChange( slug ) {
 		this.setState(
 			state => {
-				if ( includes( state.selected, slug ) ) {
+				const newSelected = state.selected;
+				const selectedIndustry = find( newSelected, { slug: slug } );
+				if ( selectedIndustry ) {
+					const newTextInputListContent = state.textInputListContent;
+					newTextInputListContent[ slug ] = selectedIndustry.detail;
 					return {
 						selected:
 							filter( state.selected, value => {
-								return value !== slug;
+								return value.slug !== slug;
 							} ) || [],
+						textInputListContent: newTextInputListContent,
```

Comment:
I wasn't able to see this state value change in my testing.

Reply:
I'm saving this value to be able to show the old input content after unchecking the option. As we remove the item from `state.selected` I save the value to not lose it if the user presses the input again.